sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -406,11 +408,18 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
         case _ => throw new IllegalArgumentException(
           s"Unsupported value type ${replaceValue.getClass.getName} ($replaceValue).")
       }
+      resolved += (resolvedColumn.name -> replaceValue)
+      // Use resolved column name onwards for filling the null values.
+      // It is needed for the column names having a dot and quoted with back-tick.
+      // Eg: "`ColWith.Dot`" will be resolved to the column with name "ColWith.Dot"
+      // in a dataframe having columns ("ColWith.Dot", "Col").
+      // If resolved name is not used, while filling null values "`ColWith.Dot`" will
+      // not match "ColWith.Dot".
```

Comment:
Usually put the comment before the code you want to comment?

```
// Use resolved column name onwards for filling the null
// ...
resolved += (resolvedColumn.name -> replaceValue)
```

Reply:
@viirya  Moved comments ahead of the statement.